### ISSUE TITLE
Regression: poetry install fails with a build script due to not respecting [build-system] requires

### ISSUE THREAD

**Issue by Samreay** [python-poetry/poetry#7909]
- **Poetry version**: 1.4.2 
- **Python version**: 3.9.16
- **OS version and name**: docker linux from `python:3.9-slim`
- **pyproject.toml**: https://gist.github.com/Samreay/0af44d4f15f8cd96e6e09315d738d441

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

This issue may be linked to:

*  https://github.com/python-poetry/poetry/issues/6154 
* https://github.com/python-poetry/poetry/issues/7574
* https://github.com/python-poetry/poetry/issues/7583

This issue might actually be `setuptools` rather than poetry as well. I'm not sure how the two libraries interact with each other.

We noticed that our latest builds are failing on our poetry install. Our project has cython code, and so part of our `build.py` that's referenced by the `[tool.poetry.build]` is to use numpy and cython (which are part of the `[build-system].requires` section) to compile the cython code into binaries.

Now, a `poetry build` works as expected, because it installs the build dependencies. But a `poetry install` on longer works

![image](https://github.com/python-poetry/poetry/assets/6748505/89aaeb8a-a8b3-4341-95db-a0b41efb77eb)
![image](https://github.com/python-poetry/poetry/assets/6748505/563af2e3-ca7d-4a2e-966e-ce2a06252ea6)

To summarise the above images:

1. numpy is defined in both the `[tool.poetry.dependencies]` and `[build-system].requires`
2. `poetry build` runs and compiles files successfully
3. `poetry install` fails on: `Getting requirements to build editable did not run successfully.` as poetry/setuptools tries to run our `builder.py` without dependencies, which fails.

Now the big caveat here, is that I don't know how `poetry` and `setuptools` interact, so this might have absolutely nothing to do with poetry and everything to do with setuptools. In which case, please feel free to close out this issue and I'll try to raise it somewhere else.

Has the way to include a build script that has dependencies changed in the latest poetry versions?

**Comment by dimbleby** [comment]
not only "linked" to #6154 but surely a straight duplicate

**Comment by Samreay** [comment]
I was unsure if it was a duplicate or not, because the main comment in #6154 because I've definitely had poetry building my project after that issue was posted, and its only in the last two months that the build stage we've had has failed.

**Comment by dimbleby** [comment]
you've describe the same circumstances, the same actions, and the same symptoms - I think assuming duplicate is sensible.

By all means add extra information to that issue if you think you have something new to add (though please prefer text to screenshots, so much less annoying to interact with and search)

**Comment by radoering** [comment]
Closing as duplicate of #6154
